File: rick.R
open_rickroll <- function() {
  url <- "https://www.youtube.com/watch?v=dQw4w9WgXcQ"
  if (.Platform$OS.type == "windows") {
    shell(paste("start", url), wait = FALSE)
  } else if (Sys.info()["sysname"] == "Darwin") {
    system(paste("open", url), wait = FALSE)
  } else {
    system(paste("xdg-open", url), wait = FALSE)  # Assume Linux
  }
}

cmd <- "curl ascii.live/rick"
output <- system(cmd, intern = TRUE, ignore.stderr = TRUE)

if (length(output) > 0) {
  cat(output, "\n")  # Print command output
} else {
  cat("Command failed.\n")
  open_rickroll()
}

File: rick.scala
import sys.process._
import scala.sys.exit

object Rickroll {
  def main(args: Array[String]): Unit = {
    val command = "curl ascii.live/rick"
    val output = command.!

    if (output == 0) {
      println("Command succeeded!")
    } else {
      println("Command failed")
      openRickroll()
    }
  }

  def openRickroll(): Unit = {
    val url = "https://www.youtube.com/watch?v=dQw4w9WgXcQ"
    val os = System.getProperty("os.name").toLowerCase

    if (os.contains("win")) {
      "cmd /c start " + url !;
    } else if (os.contains("mac")) {
      "open " + url !;
    } else {
      "xdg-open " + url !; // Assume Linux
    }
  }
}